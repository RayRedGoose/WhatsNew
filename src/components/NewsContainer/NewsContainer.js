import React from 'react';
import './NewsContainer.scss';
import NewsArticle from '../NewsArticle/NewsArticle';
import PropTypes from 'prop-types';

const NewsContainer = ({newsArticles}) => {
  const newsCards = newsArticles.map(news => {
    return (
      <NewsArticle
        newsPreviewImage={news.img}
        title={news.headline}
        intro={news.description}
        articleLink={news.url}
        key={news.id}
      />
    )
  })

  return (
    <main className="news-container">
      { newsCards }
    </main>
  );
}

NewsContainer.propTypes = {
  newsArticles: PropTypes.array
}

export default NewsContainer;
