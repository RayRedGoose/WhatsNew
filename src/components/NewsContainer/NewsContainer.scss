.news-container {
  grid-area: main;
  display: flex;
  flex-wrap: wrap;
  padding: 20px;
  justify-content: space-between;
  height: 87vh;
  overflow-y: scroll;
}

@media only screen and (max-width: 600px) {
  .news-container {
    height: 100%;
  }
}
