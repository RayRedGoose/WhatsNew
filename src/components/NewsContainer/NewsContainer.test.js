import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

it('should match the snapshot with all data passed in correctly', () => {
  const container = shallow(<NewsContainer
    newsArticles = {[{
      id: "1",
      headline: 'Mysterious comet',
      img: './image.jpg',
      description: 'Mysterious comet will cause a rare \'Unicorn\' meteor storm this week.',
      url: 'https://www.example.com'
    }]}
  />);

  expect(container).toMatchSnapshot();
});
