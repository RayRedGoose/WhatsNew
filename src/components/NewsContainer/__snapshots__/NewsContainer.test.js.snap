// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<main
  className="news-container"
>
  <NewsArticle
    articleLink="https://www.example.com"
    intro="Mysterious comet will cause a rare 'Unicorn' meteor storm this week."
    key="1"
    newsPreviewImage="./image.jpg"
    title="Mysterious comet"
  />
</main>
`;
